src/main/java/org/maxgamer/quickshop/shop/ContainerShop.java

```diff
@@ -401,19 +428,30 @@ public void delete(boolean memoryOnly) {
         if (memoryOnly) {
             // Delete it from memory
             plugin.getShopManager().removeShop(this);
+            onUnload();
         } else {
             // Delete the signs around it
             for (Sign s : this.getSigns()) {
                 s.getBlock().setType(Material.AIR);
             }
+            if (displayItem != null) {
+                displayItem.remove();
+            }
             // Delete it from the database
             // Refund if necessary
             if (plugin.getConfig().getBoolean("shop.refund")) {
-                plugin.getEconomy().deposit(this.getOwner(), plugin.getConfig().getDouble("shop.cost"), getLocation().getWorld(), getCurrency());
+                plugin.getEconomy()
+                    .deposit(this.getOwner(), plugin.getConfig().getDouble("shop.cost"),
+                        Objects.requireNonNull(getLocation().getWorld()), getCurrency());
             }
             plugin.getShopManager().removeShop(this);
+            onUnload();
```

Comment:
Why we need to call `onUnload()` for double times?

Reply:
Same as above, excellent work. My brain is too used to seeing the code so I can't even pick out duplicates like that haha.